src/main/kotlin/ecommerce/controller/CartController.kt

```diff
@@ -0,0 +1,71 @@
+package ecommerce.controller
+
+import ecommerce.annotation.LoginMember
+import ecommerce.dto.CartItemRequest
+import ecommerce.dto.CartItemResponse
+import ecommerce.dto.CartResponse
+import ecommerce.dto.RegisteredMember
+import ecommerce.model.mapper.CartItemMapper
+import ecommerce.service.CartService
+import org.springframework.data.domain.Page
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+
+@RestController
+@RequestMapping("/api/cart")
+class CartController(private val cartService: CartService) {
+    @GetMapping
+    fun getAllItems(
+        @LoginMember member: RegisteredMember,
+    ): ResponseEntity<CartResponse> {
+        val cart = cartService.findCart(member.id)
+        val body = CartResponse(cart.id, cart.items.map { CartItemMapper.toResponse(it) })
+        return ResponseEntity.status(HttpStatus.OK).body(body)
+    }
+
+    @GetMapping("/wishlist")
+    fun getAllItemsAsPages(
+        @LoginMember member: RegisteredMember,
+        @RequestParam(defaultValue = "0") page: Int,
+        @RequestParam(defaultValue = "10") size: Int,
+    ): ResponseEntity<Page<CartItemResponse>> {
+        val itemPages = cartService.getPages(member.id, page, size)
+        val headers =
+            HttpHeaders().apply {
+                add("X-Page-Number", itemPages.number.toString())
+                add("X-Page-Size", itemPages.size.toString())
+            }
+
+        return ResponseEntity.ok()
+            .headers(headers)
+            .body(itemPages)
+    }
+
+    @PostMapping
+    fun addItem(
+        @LoginMember member: RegisteredMember,
+        @RequestBody request: CartItemRequest,
+    ): ResponseEntity<CartItemResponse> {
+        val cartItem = cartService.addItem(member.id, request)
+        val response = CartItemMapper.toResponse(cartItem)
+        return ResponseEntity.created(URI("/api/${cartItem.cart.id}/items/${response.productId}")).body(response)
+    }
+
+    @DeleteMapping
+    fun deleteItem(
+        @LoginMember member: RegisteredMember,
+        @RequestBody request: CartItemRequest,
+    ): ResponseEntity<CartItemResponse?> {
+        cartService.deleteItem(member.id, request)
+        return ResponseEntity.noContent().build()
+    }
```

Comment:
```suggestion
    ): ResponseEntity<Void> {
        cartService.deleteItem(member.id, request)
        return ResponseEntity.noContent().build()
    }
```

Since this api doesn't add anything to it's response body, I think it would be better for us to reflect that in our return type.

Reply:
Thank you! The idea behind was the following. Imagine we have a CartItem with a quantity of 3. 
- we delete all pieces -> return Void
- we delete only 1 piece -> return the updated Cart with 2 pieces remaining -> return CartItem

This would represent better if only an update of the quantity was made or if everything was deleted.
It seemed to be a bit complicated to manage, so we opted for simplifying it and only returning an empty body. What do you think about this?

Reply:
If that is how our application has to work, I think that is one of the ways for us to make our api 👍 
I would consider making an `update` api for it though, and keep `delete` api as an actual removing method since changing the quantity feels more like an update to me. If me make an `update` api, that would also allow clients to add quantities more easily!